sympy/core/tests/test_constructor_postprocessor.py

```diff
@@ -0,0 +1,47 @@
+from sympy import Symbol, Mul, symbols
+
+
+class SymbolInMulOnce(Symbol):
+    # Test class for a symbol that can only appear once in a `Mul` expression.
+
+    _constructor_postprocessor_dict = {
+        "Mul": [lambda x: x],
+        "Pow": [lambda x: x.base if isinstance(x.base, SymbolInMulOnce) else x],
+        "Add": [lambda x: x],
+    }
+
+
+class SymbolRemovesOtherSymbols(Symbol):
+    # Test class for a symbol that removes other symbols in `Mul`.
+
+    def postprocess(expr):
+        args = tuple(i for i in expr.args if not isinstance(i, Symbol) or isinstance(i, SymbolRemovesOtherSymbols))
+        if args == expr.args:
+            return expr
+        return Mul.fromiter(args)
+
+    _constructor_postprocessor_dict = {
+        "Mul": [postprocess],
+    }
+
+    del postprocess
```

Comment:
@asmeurer the decorator `@staticmethod` was causing problems, this solution is way faster.